app/lib/katello/validators/content_view_puppet_module_validator.rb

```diff
@@ -3,21 +3,25 @@ module Validators
     class ContentViewPuppetModuleValidator < ActiveModel::Validator
       def validate(record)
         if record.uuid.blank? && (record.name.blank? || record.author.blank?)
-          invalid_parameters = _("Invalid puppet module parameters specified.  Either 'uuid' or 'name' and 'author' must be specified.")
-          record.errors[:base] << invalid_parameters
-          return
-        end
-
-        if record.name && record.author
-          # validate that a puppet module exists with this name+author
-          unless PuppetModule
-                  .in_repositories(record.content_view.puppet_repos)
-                  .where(:name => record.name, :author => record.author).present?
-
-            invalid_parameters = _("Puppet Module with name='%{name}' and author='%{author}' does not exist") %
-                { :name => record.name, :author => record.author }
-
-            record.errors[:base] << invalid_parameters
+          record.errors[:base] << _("Invalid puppet module parameters specified. \
+                                    Either 'uuid' or 'name' and 'author' must be specified.")
+        elsif record.name && record.author &&
+          !PuppetModule.exists?(name: record.name, author: record.author)
+          record.errors[:base] << _("Puppet Module with name='%{name}' and author='%{author}' does\
+                                    not exist") % { name: record.name, author: record.author }
+        elsif record.uuid && !PuppetModule.exists?(uuid: record.uuid)
+          record.errors[:base] << _("Puppet Module with uuid='%{uuid}' does not\
+                                    exist") % { uuid: record.uuid }
+        else
+          puppet_module = if record.uuid.blank?
+                            PuppetModule.find_by(name: record.name, author: record.author)
```

Comment:
Ahh you are right (although the API docs seem to indicate it returns the first match). However, that means that there could be following situation I believe:

Puppet Module: name=foo author=bar org=A
Puppet Module: name=foo author=bar org=B

Thus, if our content view is in Org B, and the first puppet module there is returned for Org A, the validation below would fail despite there being a puppet module that matches the criteria.